test/library/standard/Sort/errors/errors-comparator.keyandcompare.good

```diff
@@ -1,2 +1,3 @@
 errors-comparator.chpl:9: In function 'main':
+errors-comparator.chpl:17: warning: Defining a comparator with a 'key' method without implementing the keyComparator interface is deprecated. Please implement the keyComparator interface (i.e. 'record keyAndCompare: keyComparator').
```

Comment:
Should this generate both deprecations, since we aren't sure which one would be preferred by the user?  Or would that just make it harder to notice the ending error?

Reply:
If a user has defined both a key and compare but they have implemented `keyComparator`, in the non-ducktyping world that is not an error because we are not relying on a special method name.  User can implement `keyComparator` and then define `compare`, it just won't be used or recognized by `sort`.

I am not sure its worth specializing this error case to generate both deprecations, because its already an error

Reply:
But the test hasn't implemented either in this case

Reply:
This might be easier to talk about over a call

Reply:
Per our offline conversation, I've moved the deprecation check for `keyComparator` to be after the error checking for key+some other comparator method.

Done in https://github.com/chapel-lang/chapel/pull/25852/commits/1e80f13a7ec0f94a52c7f25657128f87478c56f2

Reply:
I'm only seeing test updates in that commit, shouldn't there also be a Sort.chpl update?

Reply:
Yup, it got left behind in my git buffer, the two commits are https://github.com/chapel-lang/chapel/commit/2c90272ed7d5befc2f5b4610c85d6d2603de7cae and https://github.com/chapel-lang/chapel/pull/25852/commits/734f5f11c4a76766e581cb055e8aed449e9628cf

Reply:
And now everything has been rebased so those 2 commits are one